pkg/controllers/elasticsearch/nodepool/controller.go

```diff
@@ -172,40 +172,34 @@ func (e *statefulControl) existingStatefulSet(c *v1alpha1.ElasticsearchCluster,
 	return sets[0], nil
 }
 
-func (e *statefulControl) syncPilotResources(c *v1alpha1.ElasticsearchCluster, np *v1alpha1.ElasticsearchClusterNodePool, ss *appsv1beta1.StatefulSet) error {
-	// TODO: use labels to limit which pods we list to save memory
-	allPods, err := e.podLister.Pods(c.Namespace).List(labels.Everything())
+func (e *statefulControl) syncPilotResources(c *v1alpha1.ElasticsearchCluster) error {
+	selector, err := util.SelectorForCluster(c)
+	if err != nil {
+		return err
+	}
+
+	allPods, err := e.podLister.Pods(c.Namespace).List(selector)
 	if err != nil {
 		return err
 	}
+
 	for _, pod := range allPods {
 		isMember, err := util.PodControlledByCluster(c, pod, e.statefulSetLister)
```

Comment:
`isMember` is now unused and causing compile errors.

Reply:
Ack - accidentally removed too much code!